parquet-avro/src/main/java/org/apache/parquet/avro/AvroVariantConverter.java

```diff
@@ -34,21 +35,27 @@
  * Converter for Variant values.
  */
 class AvroVariantConverter extends GroupConverter implements VariantConverters.ParentConverter<VariantBuilder> {
+  private static final Schema VARIANT_SCHEMA = SchemaBuilder.record("VariantRecord")
+      .fields()
+      .name("metadata")
+      .type()
+      .bytesType()
+      .noDefault()
+      .name("value")
+      .type()
+      .bytesType()
+      .noDefault()
+      .endRecord();
+
   private final ParentValueContainer parent;
-  private final Schema avroSchema;
   private final GenericData model;
-  private final int metadataPos;
-  private final int valuePos;
```

Comment:
Why are all of these changes needed? I think this should continue to use the schema that was passed in rather than replacing it with a different one.

Reply:
We are passing in a schema which may not have `value` or `typed_value` fields. As I understand, we want to return a record of `value` and `metadata` fields for a variant in Avro.